### DISCUSSION TITLE
File uploads very slow

### DISCUSSION THREAD

**Discussion by rishijayn** [pocketbase/pocketbase#5877] [Q&A]
Uploading a 2 MB image file to PocketBase takes 25 seconds, which feels too slow. I checked my server’s network, upload speed, and settings, but nothing seems wrong. Does anyone know why this might be happening or how to fix it?

**Comment by chnfyi** [comment]
Note that if your are using S3 storage all S3 operations are performed via the S3 APIs, meaning that all requests are proxied through your PocketBase server.

Thus, an S3 upload will always add latency compared to a simple local filesystem storage. Plus, S3 performance could vary for diverse reasons including _distance between servers, client network speed and server network speed_.

If you don't use S3 storage, please try to tell us more about how you are uploading your file to PocketBase and your server configuration (resources, reverse proxy, ...).

**Reply by rishijayn** [reply]
I was uploading image files from the pocketbase web admin itself. I tired it with and without s3 storage and there no difference because I'm using minio on the same server. I thought the problem might be the contabo vps I'm using but when i tried to do the same thing on my phone, it took only 5 seconds compared to 25 seconds avg on my pc. i tried everything but couldn't figure out what the problem is.

**Reply by chnfyi** [reply]
If after testing your network speed both on phone and desktop (e.g. _using speedtest_) you notice that there is a significant difference, so your issue is most likely networking hardware related (_different Wi-Fi cards or Wi-Fi adaptator, single vs dual-band, ..._).
